Release/5.0.0 docs (#529)
testing-library/react-hooks-testing-library: docs/api-reference.md

- [`act`](/reference/api#act)
- [`cleanup`](/reference/api#cleanup)
- [`addCleanup`](/reference/api#addcleanup)
- [`removeCleanup`](/reference/api#removecleanup)

---

## `renderHook`

```ts
function renderHook(

>   callback: (props?: any): any,

Reviewer: I don't think this syntax is valid.  This matches what you've got below for other callbacks

```suggestion
  callback: (props?: any) => any,
```

- [`act`](/reference/api#act)
- [`cleanup`](/reference/api#cleanup)
- [`addCleanup`](/reference/api#addcleanup)

> - [`removeCleanup`](/reference/api#removecleanup)

Reviewer: Whoops

---

## `renderHook`

```ts
renderHook: (
  callback: (props?: any): any,
  options?: RenderHookOptions

> ) => RenderHookResult

Reviewer: I'm not sure these read better or easier than the old ones, especially for non-TS users.

Author: > non-TS users

I think that's fair, I didn't consider that.

### `rerender`

```ts
rerender: (newProps?: any) => void
```

A function to rerender the test component, causing any hooks to be recalculated. If `newProps` are
passed, they will replace the `callback` function's `initialProps` for subsequent rerenders.

### `unmount`

```ts
unmount: () => void
```

A function to unmount the test component. This is commonly used to trigger cleanup effects for
`useEffect` hooks.

> ### `hydrate`

Reviewer: I know you've got the message about this only being available in the server renderer, but I wonder if a seperate section for SSR would make it even more apparent?

```ts
hydrate: () => void
```

> This is only used when using the `server` module. See [Rendering](/renderer) for more information

> A function to hydrate the component. This is commonly used before `act` calls and after `rerender`

Reviewer: I think it would have to be before ` rerender` too.

### `unmount`

```ts
function unmount(): void
```

A function to unmount the test component. This is commonly used to trigger cleanup effects for
`useEffect` hooks.

### `hydrate`

```ts
function hydrate(): void
```

> This is only used when using the `server` module. See [SSR](/usage/ssr-hooks) for more information
> on server-side rendering your hooks.

> A function to hydrate the component. This is required before you can interact with the hook, whether
> that is an `act` or `rerender` call. Some effects such as `useEffect` will not be called unless
> `hydrate` is called.

Reviewer: ```suggestion
A function to hydrate a server rendered component into the DOM. This is required before you can
interact with the hook, whether that is an `act` or `rerender` call. Effects created using
`useEffect` or `useLayoutEffect` are also not run on server rendered hooks until `hydrate` is
called.
```

## `act`

This is the same [`act` function](https://reactjs.org/docs/test-utils.html#act) function that is

> exported from your chosen renderer. Although, they both work the same.

Reviewer: That last sentence reads a bit odd.  I'd just drop it all together.

```suggestion
exported from your chosen renderer.
```

```

Alternatively, you can change your test to import from `@testing-library/react-hooks/pure` instead
of the regular imports. This applys to any of our export methods documented in

> [Rendering](/rendering#being-specific).

Reviewer: dead link

```suggestion
[Rendering](/installation#being-specific).
```
